src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs

```diff
@@ -1132,10 +1132,23 @@ internal override void Initialize()
                 // get the header info and the view hint
                 WideFormattingHint hint = this.InnerCommand.RetrieveFormattingHint() as WideFormattingHint;
 
+                int columnsOnTheScreen = this.InnerCommand._lo.ColumnNumber;
+                if (columnsOnTheScreen == int.MaxValue)
+                {
+                    try
+                    {
+                        columnsOnTheScreen = Console.WindowWidth;
+                    }
+                    catch
```

Comment:
`int.MaxValue` is default - can we exclude the exception?

Reply:
`OutCommandInner.TableOutputContext` class uses ` try-catch` statement assuming an environment without a console window such as implicit remoting.
Therefore, I do the same thing with `OutCommandInner.WideOutputContext` class.

>  // Tables need to use spaces for padding to maintain table look even if console window is resized. 
> // For all other output, we use int.MaxValue if the user didn't explicitly specify a width. 
>  // If we detect that int.MaxValue is used, first we try to get the current console window width. 
>  // However, if we can't read that (for example, implicit remoting has no console window), we default 
>  // to something reasonable: 120 columns.

Reply:
We already have reports of poor performance of the console. It is too costly to handle the exception for each row. My question is whether it is possible to exclude it.

Reply:
Since there are now at least two cases of this usage.  Perhaps we can have a static function get the windowwidth and if it caught an exception once, then have it return the default each time.